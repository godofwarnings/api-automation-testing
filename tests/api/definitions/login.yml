# Login test YAML definition

- test_id: "TC001_Admin_Login_Success"
  description: "Verify that an admin user can log in successfully"
  endpoint: "/auth/login"
  method: "POST"
  headers:
    Content-Type: "application/json"
  payload: '{"username": "admin", "password": "password"}'
  auth: "none"

- test_id: "TC002_Admin_Login_Failure"
  description: "Verify that login fails with an incorrect password"
  endpoint: "/auth/login"
  method: "POST"
  headers:
    Content-Type: "application/json"
  payload: '{"username": "admin", "password": "wrongpassword"}'
  auth: "none"