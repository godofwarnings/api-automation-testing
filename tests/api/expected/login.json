{
  "TC001_Admin_Login_Success": {
    "status": 200,
    "body": {
      "should_contain_key": "token"
    }
  },
  "TC002_Admin_Login_Failure": {
    "status": 403
  }
}
